src/main/java/com/google/api/codegen/util/CommonRenderingUtil.java

```diff
@@ -19,22 +19,18 @@
 import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 
 /** Utility class to process text in the templates. */
 public class CommonRenderingUtil {
-  private static Pattern singleQuoteStringPattern = Pattern.compile("'([^\\\']*)'");
-  private static Pattern doubleQuoteStringPattern = Pattern.compile("\"([^\\\"]*)\"");
 
   /** Strips the surrounding quotes from the given string */
   public static String stripQuotes(String value) {
-    Matcher singleQuoteMatcher = singleQuoteStringPattern.matcher(value);
-    Matcher doubleQuoteMatcher = doubleQuoteStringPattern.matcher(value);
-    if (singleQuoteMatcher.matches()) {
-      value = singleQuoteMatcher.group(1);
-    } else if (doubleQuoteMatcher.matches()) {
-      value = doubleQuoteMatcher.group(1);
+    for (char quoteChar : new char[] {'\'', '"'}) {
+      if (value.length() >= 2
+          && value.charAt(0) == quoteChar
+          && value.indexOf(quoteChar, 1) == value.length() - 1) {
```

Comment:
What if the string ends with an escaped quote `\"? Do we just assume that the since there's a beginning quote, the last quote ends the string and the preceding slash is just a slash?

Reply:
I think so. Our code base doesn't handle strings very well. I think it's better to keep old behavior until we understand it more.

Reply:
ie something like  `"This is my string\"`

Reply:
Yeah I'm 100% sure we'd return `This is my string \` in both the old and new implementations.

Reply:
If it's the same behavior as the old code, sgtm.

I would love to have a test to capture this behavior. Would you be able to add one?